### ISSUE TITLE
Table selection - selection is removed on paging

### ISSUE THREAD

**Issue by Sarpedion** [vuematerial/vue-material#192]
Greetings,

when I select data on a table, and then use paging to go to next page, upon returning to the page where I selected data, selection is removed.

A quick codepen demonstrating an issue:
https://codepen.io/Sarpedion/pen/pNxvPw

What I've noticed is that in data provided by selection, index always increases, even if I go to previous page.

Example, I selected first row on first page:

```
  "0": {
    "id": 1,
    "dessert": "Frozen yogurt",
    "type": "ice_cream",
    "calories": "159",
    "fat": "6.0",
    "comment": "Icy"
  }
```

I go to the next page, and then return to the first page. First row is no longer selected. When I select it again:

```
"10": {
    "id": 1,
    "dessert": "Frozen yogurt",
    "type": "ice_cream",
    "calories": "159",
    "fat": "6.0",
    "comment": "Icy"
  }
```

Note that index increased from 0 to 10.

Of course, I might be using selection and paging hooks wrong, and I do apologize if that is the case.

Thank you

**Comment by marcosmoura** [comment]
Hi @Sarpedion 

The selection will be cleared by default on page change. The index bug is ugly. I will check this.
Thanks!

**Comment by marcosmoura** [comment]
This is a breaking change. The code of pagination rely on the table indexes. If the v-for changes then the indexes refresh and add +1 to the list. This is really wrong and shouldn't be like that. I will need to rethink about this component later.
I will address to the `v0.6.0`. Sorry for the inconvenience.

**Comment by nkconnor** [comment]
any thoughts on how to fix the indexing issue? I can likely submit a PR if you point me in the right direction. thanks

**Comment by pablohpsilva** [comment]
Hello! The issue with the index was probably fixed on PR #787 , not yet merged.
Let's keep our conversation on this PR, OK?
Dudes @Sarpedion @nkconnor  thanks for this issue.
